srml/grandpa/src/lib.rs

```diff
@@ -175,6 +179,11 @@ decl_module! {
 			// FIXME: https://github.com/paritytech/substrate/issues/1112
 		}
 
+		fn on_initialize() {
+			#[cfg(feature = "migrate-authorities")]
+			Self::migrate_authorities();
```

Comment:
This migration isn't good for light clients - updated clients may ask for proof-of-read for old blocks (where value under that new well-known key has been empty). So on chains that have been started before this PR, light clients will be unable to finalize (that is only true if there have been GRANDPA authorities changes before migration, though). Not sure if we need to fix this.

Reply:
IMO, light clients should *never* check Grandpa justifications from validator sets older than the bonding duration to protect against long-range attacks. Ideally, light clients would be configured with a recent finalized block hash manually (either through software checkpoints or command-line input), then fetch the authority set from that recent block and continue.

However, this is not how Substrate works today, so maybe we do need to worry about it.

Reply:
@jimpo does that mean that light clients need to be updated at least every bonding duration?

Reply:
Only if they were not online at all during one bonding period. Basically, if a light client comes online and it's most recent epoch (or is it era? whenever it's most recent validator set was set) is more than one bonding period ago, then it shouldn't catch up without somehow being given a more recent block header by the user or by upgrading software or through some other trusted mechanism.

Reply:
late to the party, but agreed. that's part of the security model of light clients, which we haven't really launched/deployed yet anyway. breakage there is acceptable especially if it works with less bandwidth.